tinygrad/lazy.py

```diff
@@ -289,3 +281,13 @@ def elementwise_op(op:Union[UnaryOps, BinaryOps], *srcs:LazyBuffer, arg:Optional
     srcs = tuple(x.op if x.optype == BinaryOps and len(x.children) == 0 and x.realized is None else x for x in srcs)  # type: ignore
 
   return create_lazybuffer(out_device, out_shape, BinaryOps, LazyOp(op, srcs, arg), out_dtype)
+
+class _Device:
+  def __init__(self) -> None:
+    self._buffers: List[str] = [x.stem[len("ops_"):].upper() for x in (pathlib.Path(__file__).parent/"runtime").iterdir() if x.stem.startswith("ops_")]
+    try: default = "METAL" if self["METAL"] else "CUDA" if self["CUDA"] else "GPU" if self["GPU"] else "CPU"
+    except ImportError: default = "CPU"
```

Comment:
I probably need to handle cases where the dependencies are installed but device is not available. For example, if I have pycuda installed but my graphics card doesn't support CUDA.